blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx

```diff
@@ -286,12 +294,15 @@ export class RunDetailsPipeline extends Component {
                 pathArray.shift();
                 newPath = `${pathArray.join('/')}/`;
             }
+            const isParallel = this.mergedConfig.nodeReducer.edges[0] && nodeInfo.edges[0] ?
+                this.mergedConfig.nodeReducer.edges[0].id === nodeInfo.edges[0].id : false;
```

Comment:
Maybe start by extracting the above code to a function of its own? And thin in there ... break it down into human consumable fragments.